tests/unit/test_cli.py

```diff
@@ -146,3 +149,23 @@ def test_v4_parser():
     )
     actions = user_subparsers.choices["create"]._option_string_actions
     assert actions["--name"].required
+
+
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="added in 3.8")
+def test_legacy_display_without_fields_warns(fake_object_no_id):
+    printer = v4_cli.LegacyPrinter()
+
+    with mock.patch("builtins.print") as mocked:
```

Comment:
As a note `pytest` has `monkeypatch`. I'm fine with this but if you wanted to be all pytest all the time...

Reply:
Oh right, I forgot we already use that all over the place. Will change it!